test/test_torch.py

```diff
@@ -4341,6 +4341,110 @@ def test_diff(self, device, dtype):
             scalar = torch.tensor(2, device=device, dtype=dtype)
             torch.diff(scalar)
 
+    @onlyOnCPUAndCUDA
+    @dtypesIfCUDA(*set(torch.testing.get_all_math_dtypes('cuda')))
+    @dtypes(*set(torch.testing.get_all_math_dtypes('cpu')))
+    def test_gradient(self, device, dtype):
+        inputs = (
+            ((5,), [2.73], 0, 1),
+            ((5, 5), 2.0, 0, 1),
+            ((2, 2, 2), [1.17, 3.34, 1.73], [0, 1, 2], 1),
+            ((4, 4, 4), [(1., 3.0, 6., 10.), (1., 3.0, 6., 10.), (1., 3.0, 6., 10.)], [0, 1, 2], 1),
+            ((4, 4, 4), 1.7 , [0, 1, 2], 1),
+            ((4, 4, 4), 1.7 , [-2, -1], 1),
+        )
+
+        if dtype == torch.uint8:
+            return
+
+        for size, spacing, axis, edge_order in inputs:
+            t = make_tensor(size, device, dtype,)
```

Comment:
make_tensor has an option (kwargs called discontigitios) , we can make it true and it will give nontigious tensor